src/moin/storage/middleware/_tests/test_protecting.py

```diff
@@ -113,6 +115,14 @@ def test_destroy_revision(self):
         with pytest.raises(AccessDenied):
             item.destroy_revision(revid_protected)
 
+    def test_destroy_middle_revision(self):
+        item, item_name, revid0, revid1, revid2 = self._store_three_revs('joe:read,write,destroy')
```

Comment:
destroying a middle revision now requires write permission as well as destroy in order to allow for update of the parentid